data_types/src/database_name.rs

```diff
@@ -0,0 +1,147 @@
+use snafu::Snafu;
+use std::{borrow::Cow, ops::Range};
+
+/// Length constraints for a database name.
+///
+/// A `Range` is half open covering [1, 64)
+const LENGTH_CONSTRAINT: Range<usize> = 1..64;
```

Comment:
```suggestion
/// A `Range` is half open covering [1, 64]
const LENGTH_CONSTRAINT: RangeInclusive<usize> = 1..=64;
```

A maximum length of `63` characters feels a bit odd to me.

Reply:
Yeah it is weird! I took it from the [AWS bucket](https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html) docs but we don't need to worry about that I think for DB names (@pauldix ?)

Reply:
oh ok I didn't think there was precident for this 😂

Reply:
Yeah, we don't need to worry about this for db names. The structure is:
```
<bucket>/<writer id>/<database name>/...
```
The database name is part of the file path for any object inside a bucket. Your bucket would likely be called something like `iox-production` or something like that.